folium/features.py

```diff
@@ -725,93 +885,88 @@ class GeoJsonTooltip(Tooltip):
+                         u""",{{ this.tooltip_options | tojson | safe }})
+                     {% endmacro %}
+                     """)
+
+    def __init__(self, fields, aliases=None, labels=True, vegalite=None, map_key=None, data_key=None,
+                 localize=False, style=None, class_name='foliumtooltip', sticky=True, **kwargs):
         super(GeoJsonTooltip, self).__init__(
-            text='', style=style, sticky=sticky, **kwargs
+            fields=fields, aliases=aliases, labels=labels, localize=localize,
+            vegalite=vegalite, map_key=map_key, data_key=data_key, class_name=class_name,
+            style=style
         )
         self._name = 'GeoJsonTooltip'
+        kwargs.update({'sticky': sticky, 'class_name': class_name})
+        self.tooltip_options = {camelize(key): kwargs[key] for key in kwargs.keys()}
 
-        assert isinstance(fields, (list, tuple)), 'Please pass a list or ' \
-                                                  'tuple to fields.'
-        if aliases is not None:
-            assert isinstance(aliases, (list, tuple))
-            assert len(fields) == len(aliases), 'fields and aliases must have' \
-                                                ' the same length.'
-        assert isinstance(labels, bool), 'labels requires a boolean value.'
-        assert isinstance(localize, bool), 'localize must be bool.'
-        assert 'permanent' not in kwargs, 'The `permanent` option does not ' \
-                                          'work with GeoJsonTooltip.'
 
-        self.fields = fields
-        self.aliases = aliases
-        self.labels = labels
-        self.localize = localize
-        if style:
-            assert isinstance(style, str), \
-                'Pass a valid inline HTML style property string to style.'
-            # noqa outside of type checking.
-            self.style = style
+class GeoJsonPopup(GeoJsonDetail):
+    """
+    Create a popup feature to bind to each element of a GeoJson layer based on its attributes.
 
-    def warn_for_geometry_collections(self):
-        """Checks for GeoJson GeometryCollection features to warn user about incompatibility."""
-        geom_collections = [
-            feature.get('properties') if feature.get('properties') is not None else key
-            for key, feature in enumerate(self._parent.data['features'])
-            if feature['geometry']['type'] == 'GeometryCollection'
-        ]
-        if any(geom_collections):
-            warnings.warn(
-                "GeoJsonTooltip is not configured to render tooltips for GeoJson GeometryCollection geometries. "
-                "Please consider reworking these features: {} to MultiPolygon for full functionality.\n"
-                "https://tools.ietf.org/html/rfc7946#page-9".format(geom_collections), UserWarning)
+    Follows the form of GeoJsonTooltip with `fields`, `aliases` arguments to embed GeoJson properties as a table in
```

Comment:
Would be good to make the docstring selfcontained. So a user doesn't have to read the one of `GeoJsonTooltip` first.